issuing/cards.go

```diff
@@ -0,0 +1,94 @@
+package issuing
+
+import (
+	"time"
+
+	"github.com/checkout/checkout-sdk-go/common"
+)
+
+type CardType string
+
+const (
+	Physical CardType = "physical"
+	Virtual  CardType = "virtual"
+)
+
+type CardStatus string
+
+const (
+	CardActive    CardStatus = "active"
+	CardInactive  CardStatus = "inactive"
+	CardRevoked   CardStatus = "revoked"
+	CardSuspended CardStatus = "suspended"
+)
+
+type (
+	CardTypeResponse interface {
+		GetType() CardType
+		GetDetails() interface{}
+	}
+
+	PhysicalCardType struct {
+		CardDetails
+	}
+
+	VirtualCardType struct {
+		CardDetails
+		IsSingleUse bool `json:"is_single_use,omitempty"`
+	}
+
+	CardDetails struct {
+		Type             CardType        `json:"type,omitempty"`
+		Id               string          `json:"id,omitempty"`
+		CardholderId     string          `json:"cardholder_id,omitempty"`
+		CardProductId    string          `json:"card_product_id,omitempty"`
+		ClientId         string          `json:"client_id,omitempty"`
+		LastFour         string          `json:"last_four,omitempty"`
+		ExpiryMonth      int             `json:"expiry_month,omitempty"`
+		ExpiryYear       int             `json:"expiry_year,omitempty"`
+		Status           CardStatus      `json:"status,omitempty"`
+		DisplayName      string          `json:"display_name,omitempty"`
+		BillingCurrency  common.Currency `json:"billing_currency,omitempty"`
+		IssuingCountry   common.Country  `json:"issuing_country,omitempty"`
+		Reference        string          `json:"reference,omitempty"`
+		CreatedDate      *time.Time      `json:"created_date,omitempty"`
+		LastModifiedDate *time.Time      `json:"last_modified_date,omitempty"`
+	}
+
+	CardDetailsResponse struct {
+		CardTypeResponse
+	}
+)
+
+func NewPhysicalCardType() *PhysicalCardType {
+	return &PhysicalCardType{
+		CardDetails: CardDetails{Type: Physical},
+	}
+}
+
+func NewVirtualCardType() *VirtualCardType {
+	return &VirtualCardType{
+		CardDetails: CardDetails{Type: Virtual},
+	}
+}
+
+func (c *PhysicalCardType) GetType() CardType {
```

Comment:
Maybe we should declare this methods with value receivers instead of pointer receivers. Because now when we declare them as pointer receiver methods, then only pointers of the structs implement the interface.

Reply:
But it is something to refactor in all the SDK. Just leaving the comment here so that we know we have to review that in this [task](https://checkout.atlassian.net/browse/INT-1256)